crates/astria-composer/src/executor/bundle_factory/tests.rs

```diff
@@ -250,10 +303,60 @@ mod bundle_factory_tests {
         assert!(next_bundle.is_none());
     }
 
+    #[test]
+    fn pop_finished_no_longer_full() {
+        // create a bundle factory with max bundle size as 100 bytes
+        let mut bundle_factory = BundleFactory::new(100, 1);
+
+        // push a sequence action that is 100 bytes total
+        let seq_action0 = SequenceAction {
+            rollup_id: RollupId::new([0; ROLLUP_ID_LEN]),
+            data: vec![0; 100 - ROLLUP_ID_LEN],
+            fee_asset_id: default_native_asset_id(),
+        };
+        bundle_factory.try_push(seq_action0.clone()).unwrap();
+
+        // push another sequence action that is <100 bytes total to force the current bundle to
+        // flush
+        let seq_action1 = SequenceAction {
+            rollup_id: RollupId::new([1; ROLLUP_ID_LEN]),
+            data: vec![1; 100 - ROLLUP_ID_LEN],
+            fee_asset_id: default_native_asset_id(),
+        };
+        bundle_factory.try_push(seq_action1).unwrap();
+
+        // try to push a third bundle that wouldn't fit in `curr_bundle`, forcing the factory to
+        // flush it into `finished` this shouldn't work since the `finished` queue's
+        // capacity is 1.
+        let seq_action1 = SequenceAction {
+            rollup_id: RollupId::new([1; ROLLUP_ID_LEN]),
+            data: vec![1; 100 - ROLLUP_ID_LEN],
+            fee_asset_id: default_native_asset_id(),
+        };
+        let full_err = bundle_factory.try_push(seq_action1.clone());
+
+        // assert that the bundle factory has one bundle in the finished queue, that the factory is
+        // full and that err was returned
+        assert!(matches!(
+            full_err,
+            Err(BundleFactoryError::FinishedQueueFull {
+                curr_bundle_size: _,
+                seq_action: _
+            })
+        ));
```

Comment:
I think this should be removed as we tested for this above, and it's just noise now in this test.